File: app/src/test/java/org/radilabs/weather/weather/ConditionsTest.kt
package org.radilabs.weather.weather

import org.junit.Assert.assertEquals
import org.junit.Test

class ConditionsTest {
    @Test
    fun mapsKnownOpenWeatherIds() {
        assertEquals(ConditionCategory.Thunderstorm, mapOpenWeatherCondition(201))
        assertEquals(ConditionCategory.Drizzle, mapOpenWeatherCondition(300))
        assertEquals(ConditionCategory.LightRain, mapOpenWeatherCondition(500))
        assertEquals(ConditionCategory.HeavyRain, mapOpenWeatherCondition(502))
        assertEquals(ConditionCategory.Rain, mapOpenWeatherCondition(501))
        assertEquals(ConditionCategory.LightSnow, mapOpenWeatherCondition(600))
        assertEquals(ConditionCategory.Snow, mapOpenWeatherCondition(601))
        assertEquals(ConditionCategory.Fog, mapOpenWeatherCondition(741))
        assertEquals(ConditionCategory.Clear, mapOpenWeatherCondition(800))
        assertEquals(ConditionCategory.PartlyCloudy, mapOpenWeatherCondition(801))
        assertEquals(ConditionCategory.Cloudy, mapOpenWeatherCondition(803))
        assertEquals(ConditionCategory.Overcast, mapOpenWeatherCondition(804))
        assertEquals(ConditionCategory.Unknown, mapOpenWeatherCondition(999))
    }

    @Test
    fun representativeConditionUsesSeverity() {
        assertEquals(
            ConditionCategory.Thunderstorm,
            pickRepresentativeCondition(
                listOf(ConditionCategory.Clear, ConditionCategory.Thunderstorm, ConditionCategory.Rain),
            ),
        )
    }

    @Test
    fun airQualityScaleIsOpenWeatherNotEpa() {
        assertEquals(AirQualityCategory.Fair, airQualityCategory(2))
        assertEquals(AirQualityCategory.Unknown, airQualityCategory(9))
    }
}

File: tests/weather/conditions.test.ts
import { describe, expect, it } from "vitest";
import { mapOpenWeatherCondition, pickRepresentativeCondition } from "../../src/weather/conditions";

describe("mapOpenWeatherCondition", () => {
  it("maps known OpenWeather ids deterministically", () => {
    expect(mapOpenWeatherCondition(800)).toBe("clear");
    expect(mapOpenWeatherCondition(801)).toBe("partly-cloudy");
    expect(mapOpenWeatherCondition(803)).toBe("cloudy");
    expect(mapOpenWeatherCondition(804)).toBe("overcast");
    expect(mapOpenWeatherCondition(300)).toBe("drizzle");
    expect(mapOpenWeatherCondition(500)).toBe("light-rain");
    expect(mapOpenWeatherCondition(501)).toBe("rain");
    expect(mapOpenWeatherCondition(502)).toBe("heavy-rain");
    expect(mapOpenWeatherCondition(202)).toBe("thunderstorm");
    expect(mapOpenWeatherCondition(600)).toBe("light-snow");
    expect(mapOpenWeatherCondition(601)).toBe("snow");
    expect(mapOpenWeatherCondition(741)).toBe("fog");
  });

  it("falls back safely for unknown codes", () => {
    expect(mapOpenWeatherCondition(9999)).toBe("unknown");
  });
});

describe("pickRepresentativeCondition", () => {
  it("selects the most severe category in the day", () => {
    expect(
      pickRepresentativeCondition(["clear", "rain", "partly-cloudy"]),
    ).toBe("rain");
  });
});
